src/main/scala/it/pagopa/interop/purposeprocess/api/impl/RiskAnalysisValidation.scala

```diff
@@ -21,9 +23,25 @@ object RiskAnalysisValidation {
     * @return Validated risk analysis
     */
   def validate(form: RiskAnalysisForm): ValidationResult[RiskAnalysisFormSeed] = {
-    val answersJson: JsObject = form.answers.toJson.asJsObject
 
-    val validations = validateForm(answersJson, validationRules)
+    val sanitizedForm = form.copy(answers = form.answers.filter(_._2.nonEmpty))
+
+    val validationRules: ValidationResult[List[ValidationEntry]] =
+      RiskAnalysisService.riskAnalysisForms
+        .get(sanitizedForm.version)
+        .fold[ValidationResult[List[ValidationEntry]]](UnexpectedTemplateVersion(sanitizedForm.version).invalidNec)(
```

Comment:
Passing types to `in/validNec` should avoid the need to declare the type of the fold, innit?

Reply:
In this case the refactor could be
```
.fold(UnexpectedTemplateVersion(sanitizedForm.version).invalidNec[List[ValidationEntry]])(
    configsToRules(_).validNec[UnexpectedTemplateVersion]
)
```
I just have a couple of doubts: 
- formally this function could return the generic `RiskAnalysisValidationError` type, but fixing types in `in/validNec` means using the specific `UnexpectedTemplateVersion` because of the `fold` invariance. Am I missing something here?
- we are actually adding more characters 🤣

Reply:
>     * formally this function could return the generic `RiskAnalysisValidationError` type, but fixing types in `in/validNec` means using the specific `UnexpectedTemplateVersion` because of the `fold` invariance. Am I missing something here?

You're not. But is it a problem? As far as I can see that is the only error that might be returned in this fold.

> 
>     * we are actually adding more characters 🤣

You're right here, let's leave it like it is so